econml/sklearn_extensions/ensemble.py

```diff
@@ -65,14 +59,36 @@ def _parallel_add_trees(tree, forest, X, y, sample_weight, s_inds, tree_idx, n_t
     # Calculate the total weight of estimation samples on each tree node:
     # \sum_i sample_weight[i] * 1{i \\in node}
     weight_est = scipy.sparse.csr_matrix(
-        sample_weight_est.reshape(1, -1)).dot(path_est)
+        sample_weight_est.reshape(1, -1)).dot(path_est).todense()
```

Comment:
So `path_est` is sparse, but `sample_weight_est` is initially dense and so we first make it sparse to compute the dot product, and then densify the result?  It looks like there's an existing scipy function [safe_sparse_dot](https://scikit-learn.org/stable/modules/generated/sklearn.utils.extmath.safe_sparse_dot.html) that does a perhaps more efficient version of that already, and [the implementation](https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/utils/extmath.py#L119) just boils down to `a @ b` for the not-higher-than-2-dimensional case, which I assume we're in here.

Reply:
Cool! Yeah this is much cleaner.